tests/core/templates/test_oa_hash_map.h

```diff
@@ -0,0 +1,193 @@
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_OA_HASH_MAP_H
+#define TEST_OA_HASH_MAP_H
+
+#include "core/templates/oa_hash_map.h"
+#include "scene/resources/texture.h"
+
+#include "tests/test_macros.h"
+
+namespace TestOAHashMap {
+
+TEST_CASE("[OAHashMap] Insert element") {
+	OAHashMap<int, int> map;
+	map.insert(42, 84);
+	int data = 0;
+	bool lookup_res = map.lookup(42, data);
+	int value = *map.lookup_ptr(42);
+	CHECK(lookup_res);
+	CHECK(value == 84);
+	CHECK(data == 84);
+}
+
+TEST_CASE("[OAHashMap] Set element") {
+	OAHashMap<int, int> map;
+	map.set(42, 84);
+	int data = 0;
+	bool lookup_res = map.lookup(42, data);
+	int value = *map.lookup_ptr(42);
+	CHECK(lookup_res);
+	CHECK(value == 84);
+	CHECK(data == 84);
+}
+
+TEST_CASE("[OAHashMap] Overwrite element") {
+	OAHashMap<int, int> map;
+	map.set(42, 84);
+	map.set(42, 1234);
+	int result = *map.lookup_ptr(42);
+	CHECK(result);
```

Comment:
This should check the new value as well:

```suggestion
	CHECK(result == 1234);
```